core/trino-spi/src/main/java/io/trino/spi/type/LongDecimalType.java

```diff
@@ -152,7 +153,7 @@ private static long xxHash64Operator(@BlockPosition Block block, @BlockIndex int
 
     private static long xxHash64(long low, long high)
     {
-        return XxHash64.hash(low) ^ XxHash64.hash(high);
+        return HashUtils.mix(XxHash64.hash(low), XxHash64.hash(high));
```

Comment:
I would inline the mixing code here.  Also, I would talk to @martint  to see if we can add a method to XxHash64 that takes a 128bit number (as two longs), and produces the proper xxhash64 result for that.

Reply:
Agree. `mix` method in xxHash64 could be public.